test/files/run/settings-parse.scala

```diff
@@ -3,9 +3,8 @@ import scala.language.postfixOps
 import scala.tools.nsc._
 
 object Test {
-  val tokens        = List("", "-deprecation", "foo.scala")
-  val subsets       = tokens.toSet.subsets.toList
-  val permutations0 = subsets.flatMap(_.toList.permutations).distinct
+  val tokens        = "" :: "-deprecation" :: "foo.scala" :: Nil
+  val permutations0 = tokens.toSet.subsets.flatMap(_.toList.permutations).toList.distinct
```

Comment:
This would probably be cleaner if there was a "STOP" token and the computation was `.permutations.map(_ takeWhile(_ != "STOP")).toList.distinct`, but this gets the job done.